src/ai/ai.cpp

```diff
@@ -4366,7 +4438,7 @@ void assign_targets(sys::state& state, dcon::nation_id n) {
 		if(!potential_targets[i].location)
 			continue; // target has been removed as too close by some earlier iteration
 
-		auto target_attack_force = estimate_attack_force(state, potential_targets[i].location, n);
```

Comment:
Here and in the location below, for example, this function may be called repeatedly on basically the same provinces. Is it necessary to cache the result now?

Reply:
cached